tests/Functional/Schema/SchemaManagerFunctionalTestCase.php

```diff
@@ -1418,6 +1419,58 @@ public function testCommentInTable(): void
         $table = $this->schemaManager->listTableDetails('table_with_comment');
         self::assertSame('Foo with control characters \'\\', $table->getComment());
     }
+
+    public function testCreatedCompositeForeignKeyOrderIsCorrectAfterCreation(): void
+    {
+        $localColumns = ['child_col2', 'child_col1'];
+        $foreignColumns = ['col2', 'col1'];
+
+        $table = new Table('test_table_foreign_key_referenced');
+
+        $table->addOption('collate', 'utf8_general_ci');
+        $table->addOption('charset', 'utf8');
+        $table->addOption('comment', 'Test foreign key referenced table');
+
+        $table->addColumn('col1', Types::INTEGER, ['notNull' => true]);
+        $table->addColumn('col2', Types::INTEGER, ['notNull' => true]);
+        $table->setPrimaryKey($foreignColumns);
+
+        $this->schemaManager->dropAndCreateTable($table);
```

Comment:
I think `dropAndCreate()` won't work well for the tables connected by a foreign key. Please try running this test as follows  and make sure that both runs pass:
```shell
phpunit \
  --repeat 2 \
  --filter testCreatedCompositeForeignKeyOrderIsCorrectAfterCreation \
  tests/Functional/Schema/MySQLSchemaManagerTest.php
```

You may need to use something like this instead: https://github.com/doctrine/dbal/blob/81c54e8caf8b959609fa6dab244904c54f0181fc/tests/Functional/Schema/OracleSchemaManagerTest.php#L128-L132

For that, you'll need to temporarily comment out this line which is another PITA:
https://github.com/doctrine/dbal/blob/81c54e8caf8b959609fa6dab244904c54f0181fc/tests/Functional/Schema/MySQLSchemaManagerTest.php#L21-L21

Reply:
@morozov Thanks for the hint. However, dropping did not help. I had to first remove the foreign key, otherwise it did not work. Probably could have kept the dropAndCreate now but I have kept them separate now.